tests/test_totem.lua

```diff
@@ -112,6 +113,39 @@ function tests.test_assertTable()
   end
 end
 
+function tests.test_assertEqRet()
+  -- Make a local tester
+  local retTester = totem.Tester()
+  local retTests = totem.TestSuite()
+
+  -- This test should produce 2 assertions and 1 failure
+  function retTests.dummyFn()
+    -- When the 'ret' value is set to true there should be no assertion
+    -- regardless of the outcome of the equality 
+    retTester:eq({1}, {1}, '', 0, true)
+    retTester:eq({1}, {2}, '', 0, true)
+
+    -- On a truthful equality test, with a 'ret' value of false, there should
+    -- be one assertion but no failure
+    retTester:eq({1}, {1}, '', 0, false)
+
+    -- On a non truthful equality test, with a 'ret' value of false, there
+    -- should be one assertion and one failure
+    retTester:eq({1}, {2}, '', 0, false)
+  end
+
+  -- Run the retTester in pcall so that it does not print anything
+  retTester:add(retTests)
+  pcall(retTester.run, retTester)
```

Comment:
This will not work the way you intend it to - the printing will happen as print or rather io.write commands, not because of a thrown error, so pcall won't prevent output here. You could do this by wrapping the call to retTester.run with 
```lua
local oldWrite, io.write = io.write, function() end
-- do stuff here which now won't produce output
io.write = oldWrite
```